File: Models/PriceAlert.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{
    public class PriceAlert
    {
        public int Id { get; set; }
        
        [Required]
        public string AppUserId { get; set; } = string.Empty;
        
        [Required]
        public int StockId { get; set; }
        
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TargetPrice { get; set; }
        
        [Required]
        public AlertType AlertType { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime? TriggeredAt { get; set; }
        
        // Navigation properties
        public AppUser AppUser { get; set; } = null!;
        public Stock Stock { get; set; } = null!;
    }
    
    public enum AlertType
    {
        PriceAbove,
        PriceBelow
    }
}

File: src/main/java/com/stockwatch/stockwatchpro/models/PriceAlert.java
package com.stockwatch.stockwatchpro.models;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "price_alerts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PriceAlert {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "app_user_id", nullable = false)
    private String appUserId;

    @Column(name = "stock_id", nullable = false)
    private Integer stockId;

    @Column(name = "target_price", nullable = false, precision = 18, scale = 2)
    private BigDecimal targetPrice;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AlertType alertType;

    @Column(name = "is_active")
    private Boolean isActive = true;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "triggered_at")
    private LocalDateTime triggeredAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "app_user_id", insertable = false, updatable = false)
    private AppUser appUser;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "stock_id", insertable = false, updatable = false)
    private Stock stock;

    public enum AlertType {
        PRICE_ABOVE,
        PRICE_BELOW
    }
}
